cmd/testpeers.go

```diff
@@ -204,14 +204,14 @@ func pingPeerOnce(ctx context.Context, tcpNode host.Host, peer p2p.Peer) (ping.R
 
 func pingPeerContinuously(ctx context.Context, tcpNode host.Host, peer p2p.Peer, resCh chan ping.Result) {
 	for {
+		r, err := pingPeerOnce(ctx, tcpNode, peer)
+		if err != nil {
+			return
+		}
 		select {
 		case <-ctx.Done():
```

Comment:
This appears to be useless given that `pingPeerOnce` must return when `ctx.Done()`.

Reply:
That's true and this context doneness check is triggered regularly on `pingPeerOnce` (I see it being triggered on every `pingLoad` test). However, I have observed the code panic-ing, because `resCh` was closed and it tried sending data to it. I think it is because there is canceling of the context and closing of the channel happening between `r, err := pingPeerOnce(ctx, tcpNode, peer)` and `resCh <- r` (which is the error check). I've observed it happening couple of times, but it's really rare to catch.

Reply:
That's why the channel shall normally to be closed by Producer.

Reply:
Pushed new commit where I create separate channels for each `pingPeerContinuously` call and close it upon exiting `pingPeerContinuously` function.

Reply:
In this case you could simply use `sync.WaitGroup` to ensure all spawned goroutines will stop writing and then close the single `testResCh` after `.Wait()`. 
A few other observations:
* You may not need `deadlineC` at all, if you use a context with deadline with will give you a cancel function.
* You may not need `finished` flag at all, if you simply exit the infinite loop and then wait on the mentioned `WaitGroup`.